spec/services/saml_request_validator_spec.rb

```diff
@@ -2,53 +2,59 @@
 
 RSpec.describe SamlRequestValidator do
   describe '#call' do
-    context 'valid authn context and sp and authorized nameID format' do
-      it 'returns FormResponse with success: true' do
-        sp = ServiceProvider.find_by(issuer: 'http://localhost:3000')
-        authn_context = [Saml::Idp::Constants::IAL1_AUTHN_CONTEXT_CLASSREF]
-        name_id_format = Saml::Idp::Constants::NAME_ID_FORMAT_PERSISTENT
-        extra = {
-          authn_context: authn_context,
-          service_provider: sp.issuer,
-          nameid_format: name_id_format,
-          authn_context_comparison: 'exact',
-        }
+    let(:sp) { ServiceProvider.find_by(issuer: 'http://localhost:3000') }
+    let(:name_id_format) { Saml::Idp::Constants::NAME_ID_FORMAT_PERSISTENT }
+    let(:authn_context) { [Saml::Idp::Constants::IAL1_AUTHN_CONTEXT_CLASSREF] }
+    let(:comparison) { 'exact' }
+    let(:extra) do
+      {
+        authn_context: authn_context,
+        service_provider: sp&.issuer,
+        nameid_format: name_id_format,
+        authn_context_comparison: comparison,
+      }
+    end
 
-        response = SamlRequestValidator.new.call(
-          service_provider: sp,
-          authn_context: authn_context,
-          nameid_format: name_id_format,
-        )
+    let(:response) do
+      SamlRequestValidator.new.call(
+        service_provider: sp,
+        authn_context: authn_context,
+        nameid_format: name_id_format,
+      )
+    end
 
+    context 'valid authn context and sp and authorized nameID format' do
+      it 'returns FormResponse with success: true' do
         expect(response.to_h).to include(
           success: true,
           errors: {},
           **extra,
         )
       end
+
```

Comment:
this file is noisy bc i updated the test file to reduce duplication/make it clearer what was being tested. 

i introduced tests on line 35 and line 200 specific to IALMAX